python/lsst/ts/electrometer/controller.py

```diff
@@ -333,47 +360,115 @@ async def start_scan_dt(self, scan_duration):
+        primary_hdu.header["LOCATN"] = (self.location, "Location of Instrument")
+        primary_hdu.header["ORIGIN"] = (
+            self.csc.salinfo.name,
+            "Name of the program that produced this data.",
+        )
+        primary_hdu.header["DATE-BEG"] = (
+            astropy.time.Time(self.manual_start_time, format="unix_tai").tai.fits,
+            "When start scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["DATE-END"] = (
+            astropy.time.Time(self.manual_end_time, format="unix_tai").tai.fits,
+            "When stop scan command sent to CSC (TAI)",
+        )
+        primary_hdu.header["TIMESYS"] = ("TAI", "Format of timestamps")
+        primary_hdu.header["SCANTIME"] = (self.scan_duration, "Duration of scan [s]")
+        primary_hdu.header["SAMPTIME"] = (
+            self.integration_time,
+            "Duration of each sample [s]",
+        )
+        primary_hdu.header["FILTMED"] = (
+            self.median_filter_active,
+            "Median Filter Active",
+        )
+        primary_hdu.header["FILTAVG"] = (
+            self.avg_filter_active,
+            "Average Filter Active",
+        )
+        primary_hdu.header["IMGTYPE"] = (
+            self.mode,
+            "Options are charge, voltage, current",
+        )
+        primary_hdu.header["SENSBRND"] = (self.sensor_brand, "Sensor brand")
+        primary_hdu.header["SENSMODL"] = (self.sensor_model, "Sensor model")
+        primary_hdu.header["SERIAL"] = (self.sensor_serial, "Sensor serial number")
+        primary_hdu.header["TEMP"] = (
+            self.temperature,
+            "Measurement from probe if attached and declared (Celcius)",
+        )
+        primary_hdu.header["VSOURCE"] = (
+            self.vsource,
+            "Voltage input if active and attached",
+        )
+        return primary_hdu
+
+    async def write_fits_file(self, signal, times, temperature, unit, voltage):
         """Write fits file of the intensity, time, and temperature values.
 
         Parameters
         ----------
-        intensity : `list`
-            The intensity values
+        signal : `list`
+            The signal values.
         times : `list`
-            The time of the intensity value
+            The time of the signal data taken.
         temperature : `list`
-            The temperature of the intensity value
+            The temperature of the signal data. (constant)
         unit : `list`
-            The unit of the intensity value.
+            The unit of the signal data. (constant)
+        voltage : `list`
+            The voltage source input. (constant)
         """
-        data = np.array([times, intensity])
-        hdu = fits.PrimaryHDU(data)
-        hdr = hdu.header
-        hdr["CLMN1"] = ("Time", "Time in seconds")
-        hdr["CLMN2"] = "Intensity"
+        if self.temperature_attached:
+            self.temperature = temperature[0]
```

Comment:
Why are there multiple temperature (and voltage) values, why only use the first value of the list and not all of them?

Reply:
There are multiple temperature and voltage input values as part of each reason but the reason I use only one is because those values don't change across each reading.

Reply:
Please add a comment in the code explaining that.